go/vt/vttablet/tabletserver/repltracker/writer.go

```diff
@@ -133,34 +163,52 @@ func (w *heartbeatWriter) Open() {
 	// block this thread, and we could end up in a deadlock.
 	// Instead, we try creating the database and table in each tick which runs in a go routine
 	// keeping us safe from hanging the main thread.
-	w.appPool.Open(w.env.Config().DB.AppWithDB())
-	w.allPrivsPool.Open(w.env.Config().DB.AllPrivsWithDB())
-	if w.onDemandDuration == 0 {
-		w.enableWrites(true)
-		// when onDemandDuration > 0 we only enable writes per request
-	} else {
+
+	if w.keyspaceShard != testKeyspaceShard {
+		// testKeyspaceShard indicates we're running in a unit test, in which case mock
+		// pools will have been opened.
+		w.appPool.Open(w.env.Config().DB.AppWithDB())
+		w.allPrivsPool.Open(w.env.Config().DB.AllPrivsWithDB())
+	}
```

Comment:
I think that we should find another way to do this. I say that as it's perfectly fine and acceptable for users to create an unsharded keyspace named `test`. We may (if not already) be doing that in endtoend tests too.

Reply:
That makes sense. I wasn't fully happy with this.